sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import-business-validation.service.ts

```diff
@@ -0,0 +1,149 @@
+        ApplicationBulkWithdrawalImportBusinessValidationModel,
+        model,
+        {
+          enableImplicitConversion: true,
+        },
+      );
+      const validationsErrors = validateSync(
+        applicationBulkWithdrawalBusinessValidationModel,
+      );
+      const allErrors: string[] = [];
+      const allWarnings: ValidationResult[] = [];
+      validationsErrors.forEach((error) => {
+        const validation = this.extractValidationsByType(error);
+        allErrors.push(...validation.errors);
+        allWarnings.push(...validation.warnings);
+      });
+
+      if (!silently && !!allErrors.length) {
+        throw new CustomNamedError(
+          "The application bulk withdrawal has critical errors.",
+          APPLICATION_BULK_WITHDRAWAL_VALIDATION_CRITICAL_ERROR,
+          allErrors,
+        );
+      }
+
+      return {
+        index,
+        applicationBulkWithdrawalBusinessValidationModel,
+        warnings: allWarnings,
+        errors: allErrors,
+      };
+    });
+  }
+
+  /**
+   * Verify if all application withdrawal business validations were performed with success and throw
+   * a BadRequestException in case of some failure.
+   * @param businessValidationsResult business validation results to be verified.
+   */
+  assertBusinessValidationsAreValid(
+    businessValidationsResult: ApplicationWithdrawalValidationResult[],
+  ) {
+    const businessValidations = businessValidationsResult.filter(
+      (businessValidationResult) =>
+        businessValidationResult.errors.length ||
+        businessValidationResult.warnings.length,
+    );
+    if (businessValidations.length) {
+      // At least one error or warning was detected.
+      const errorValidationResults: ApplicationBulkWithdrawalValidationResultAPIOutDTO[] =
+        businessValidations.map((validation) => ({
+          recordIndex: validation.index,
+          applicationNumber:
+            validation.applicationBulkWithdrawalBusinessValidationModel
+              .applicationNumber,
+          withdrawalDate:
+            validation.applicationBulkWithdrawalBusinessValidationModel
+              .withdrawalDate,
+          sin: validation.applicationBulkWithdrawalBusinessValidationModel.sin,
+          errors: validation.errors,
+          warnings: validation.warnings,
+          infos: [],
+        }));
+      throw new BadRequestException(
+        new ApiProcessError(
+          "One or more fields have validation errors.",
+          APPLICATION_WITHDRAWAL_BUSINESS_VALIDATION_ERROR,
+          errorValidationResults,
+        ),
+      );
+    }
+  }
+
+  /**
+   * Inspect the validation error and its children when an error
+   * happen as it can have an additional context (must be considered a warning)
+   * or it is a critical error (has no warning context).
+   * @param error error to be inspected.
+   * @returns errors and warnings.
+   */
+  private extractValidationsByType(error: ValidationError): {
```

Comment:
Is there a reason to have this method duplicated instead of centralized and reused for offerings and withdrawals?